src/diffusers/pipelines/flux/pipeline_flux.py

```diff
@@ -375,7 +375,9 @@ def encode_prompt(
                 # Retrieve the original scale by scaling back the LoRA layers
                 unscale_lora_layers(self.text_encoder_2, lora_scale)
 
-        text_ids = torch.zeros(batch_size, prompt_embeds.shape[1], 3).to(device=device, dtype=self.text_encoder.dtype)
+        dtype = self.text_encoder.dtype if self.text_encoder is not None else self.transformer.dtype
+        text_ids = torch.zeros(batch_size, prompt_embeds.shape[1], 3).to(device=device, dtype=dtype)
```

Comment:
The `dtype` change had to be made to allow tests that check `prompt_embeds`. This has an overlap with https://github.com/huggingface/diffusers/pull/9049. Can close https://github.com/huggingface/diffusers/pull/9049 to fix it here?